version: "3.11"

services:
  api:
    platform: linux/amd64
    build:
      context: .
    image: rafacom/test_docker_compose_1
    ports:
      - "80:80"
    volumes:
      - ./docker_config.py:/code/default_config.py
    depends_on:
      pg:
        condition: service_healthy
      migrations:
        condition: service_completed_successfully
  migrations:
    build:
      context: .
    image: rafacom/test_docker_compose_1
    volumes:
      - ./docker_config.py:/code/default_config.py
    platform: linux/amd64
    depends_on:
      pg:
        condition: service_healthy
    command: flask db upgrade
  pg:
    image: postgres:latest
    environment:
      POSTGRES_USER: flask_app
      POSTGRES_PASSWORD: flask_app_password
      POSTGRES_DB: flask_app
    volumes:
      - ./pg_data:/var/lib/postgresql/data
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U flask_app"]
      interval: 5s
      timeout: 5s
      retries: 5